cmd/gctsDeployCommit.go

```diff
@@ -0,0 +1,82 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"net/http/cookiejar"
+
+	gabs "github.com/Jeffail/gabs/v2"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func gctsDeployCommit(config gctsDeployCommitOptions, telemetryData *telemetry.CustomData) {
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+	httpClient := &piperhttp.Client{}
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := deployCommit(&config, telemetryData, &c, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func deployCommit(config *gctsDeployCommitOptions, telemetryData *telemetry.CustomData, command execRunner, httpClient piperhttp.Sender) error {
+
+	cookieJar, cookieErr := cookiejar.New(nil)
+	if cookieErr != nil {
+		return errors.Wrap(cookieErr, "deploy commit failed")
+	}
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+		Username:  config.Username,
+		Password:  config.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	url := config.Host +
+		"/sap/bc/cts_abapvcs/repository/" + config.Repository +
+		"/pullByCommit?sap-client=" + config.Client
+	if config.Commit != "" {
+		url = url + "&request=" + config.Commit
```

Comment:
What can technically be provided here. Is it only a 40 digit commit hash? Can it be any treeish? In the later case it could also be a branch. Branches/tags can contain slashes, e.g. `releases/1.2.3`. That slash needs to be url escaped (`%2F`), applies also for the dots. In case we can have branches or tags here I would like suggest to url-encode the commit. Maybe the naming can be improved in order to outline with the name what exactly can be provided.

Reply:
... with encoding we are on the safe side.

Reply:
I was asked to clarify what I mean with the encoding here since

> url-encode the commit

is not very precise.

- In the coding above we refer to a commit with parameter `request`.
- There are in general several ways to refer to a commit
  - the hash
  - a tag pointing to the commit
  - a branch - in case the commit it the HEAD of a branch.
- Since "our commit"  is part of the url it needs to be url encoded

In case we have a tag or branch `releases/1.2.3` this needs to be `https://example.org/myPath?request=releases%2F/1%2E2%2E3` rather than `https://example.org/myPath?request=releases/1.2.3`.

Reply:
So far it can be just an Commit hash but I agree with Marcus. An URL encoding would be a nice solution in that position.